### PR TITLE
Avoid type conversion when encoding MeasureSchema

### PR DESCRIPTION
**Pull request by yhwangBonc**

### CODE REVIEW

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/CompressionType.java [hunk 1]
```diff
@@ -52,6 +52,35 @@ public static CompressionType deserialize(short i) {
     }
   }
 
+  /**
+   * give an byte to return a compression type.
+   *
+   * @param i byte number
+   * @return CompressionType
+   */
+  public static CompressionType byteToEnum(byte i) {
```
Comment (mychaow) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/CompressionType.java:61]: byte i， why call this name

**Review by mychaow — CHANGES_REQUESTED** [review]

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/CompressionType.java (hunk 1 above)
Comment (yhwangBonc) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/CompressionType.java:61]: It's my negligence. I'll change it

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSEncoding.java [hunk 2]
```diff
@@ -51,6 +51,35 @@ public static TSEncoding deserialize(short i) {
...
   }
 
+  /**
+   * give an byte to return a encoding type.
+   *
+   * @param i byte number
+   * @return encoding type
+   */
+  public static TSEncoding byteToEnum(byte i) {
+    switch (i) {
```
Comment (neuyilan) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSEncoding.java:61]: better to check the size of this parameter as TSDataType

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/schema/MeasurementSchema.java [hunk 3]
```diff
@@ -98,12 +98,11 @@ public static MeasurementSchema deserializeFrom(InputStream inputStream) throws
 
     measurementSchema.measurementId = ReadWriteIOUtils.readString(inputStream);
```
Comment (neuyilan) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/write/schema/MeasurementSchema.java:99]: Add one UT of the serialize and deserialize method

**Review by neuyilan — CHANGES_REQUESTED** [review]

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSEncoding.java (hunk 2 above)
Comment (yhwangBonc) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/enums/TSEncoding.java:61]: Copy that. I'll change it

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/schema/MeasurementSchema.java (hunk 3 above)
Comment (yhwangBonc) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/write/schema/MeasurementSchema.java:99]: OK, I'll add it

File: tsfile/src/test/java/org/apache/iotdb/tsfile/write/writer/MeasurementSchemaSerializeTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package org.apache.iotdb.tsfile.write.writer;
```
Comment (HTHou) [inline tsfile/src/test/java/org/apache/iotdb/tsfile/write/writer/MeasurementSchemaSerializeTest.java:1]: Please add the Apache header.

File: tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java [hunk 5]
```diff
@@ -82,6 +83,19 @@ public static boolean readBool(ByteBuffer buffer) {
     return a == 1;
   }
 
+  /**
+   * read a byte from inputStream.
+   */
+  public static byte readByte(InputStream inputStream) throws IOException {
```
Comment (jixuan1989) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java:89]: hm.. this method is overdesigned..

If you read `InputStream.read(byte[])` method, you will find it actually just call `byte c = (byte)read()`;
So, just `return (byte) read()` is ok.

**Review by jixuan1989 — COMMENTED** [review]
better not change the serialize and deserialize in `MeasurementSchema` for version compatibility.

File: tsfile/src/test/java/org/apache/iotdb/tsfile/write/writer/MeasurementSchemaSerializeTest.java (hunk 4 above)
Comment (yhwangBonc) [reply tsfile/src/test/java/org/apache/iotdb/tsfile/write/writer/MeasurementSchemaSerializeTest.java:19]: copy that

File: tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java (hunk 5 above)
Comment (yhwangBonc) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java]: OK, thanks for your guidance

**Review by mychaow — APPROVED** [review]

**Review by qiaojialin — APPROVED** [review]
Good work!

**Pull request merged**